core/src/main/java/org/mule/runtime/core/internal/util/rx/ReactorTransactionUtils.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.internal.util.rx;
+
+import reactor.util.context.Context;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.function.Function;
+
+import static java.util.Collections.emptyList;
+
+/**
+ * Utils class to simulate transactional behavior in reactor.
+ *
+ * @since 4.5, 4.4.1, 4.3.1
+ */
+public class ReactorTransactionUtils {
+
+  public static final String TX_SCOPES_KEY = "mule.tx.activeTransactionsInReactorChain";
+
+  public static boolean isTxActiveInContext(Context ctx) {
```

Comment:
i have concerns with the name, the tx is not active in the context, the context just indicates that a tx will be active when the chain is executed.